From 5edb19e907ad2a8ea9f1a446d8c474749f343e57 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Fri, 31 May 2013 22:53:50 +0530
Subject: [PATCH 33/56] During playback of ALAC5.1 and multichannel wave files
 Audio is not heard on IHF

BZ: 57613

This patch contains changes in AudioMixer.cpp. Currently mutli-channel SRC
isn't added, so change is done to use stereo when multi-channel downmix
buffer is present.

Change-Id: Ib8fc5b01d3679d7df2838fc22e8296a2aaa139b3
Orig-Change-Id: Ia002e4240211e690bffb82d6de4341bcd9a3117a
Category: aosp improvement
Domain: AUDIO_MEDIA-Framework
Origin: internal
Upstream-Candidate: no, needsrework
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/AudioMixer.cpp | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/services/audioflinger/AudioMixer.cpp b/services/audioflinger/AudioMixer.cpp
index f9f10e6..a27f142 100644
--- a/services/audioflinger/AudioMixer.cpp
+++ b/services/audioflinger/AudioMixer.cpp
@@ -552,13 +552,16 @@ bool AudioMixer::track_t::setResampler(uint32_t value, uint32_t devSampleRate)
     if (value != devSampleRate || resampler != NULL) {
         if (sampleRate != value) {
             sampleRate = value;
+
+            uint8_t channels = downmixerBufferProvider != NULL ? MAX_NUM_CHANNELS : channelCount;
+
             if (resampler == NULL) {
 #ifdef USE_INTEL_RSP
                 if (AudioResamplerIA::sampleRateSupported(sampleRate, devSampleRate)) {
-                    resampler = AudioResampler::create(format, channelCount,
+                    resampler = AudioResampler::create(format, channels,
                                 devSampleRate, AudioResampler::INTEL_HIGH_QUALITY);
                 } else {
-                    resampler = AudioResampler::create(format, channelCount, devSampleRate);
+                    resampler = AudioResampler::create(format, channels, devSampleRate);
                 }
 #else
                 ALOGV("creating resampler from track %d Hz to device %d Hz", value, devSampleRate);
@@ -579,9 +582,7 @@ bool AudioMixer::track_t::setResampler(uint32_t value, uint32_t devSampleRate)
                     quality = AudioResampler::DEFAULT_QUALITY;
                 }
                 resampler = AudioResampler::create(
-                        format,
-                        // the resampler sees the number of channels after the downmixer, if any
-                        downmixerBufferProvider != NULL ? MAX_NUM_CHANNELS : channelCount,
+                        format, channels,
                         devSampleRate, quality);
                 resampler->setLocalTimeFreq(sLocalTimeFreq);
 #endif
@@ -591,7 +592,8 @@ bool AudioMixer::track_t::setResampler(uint32_t value, uint32_t devSampleRate)
                 if (resampler->mResType == AudioResampler::INTEL_SRC &&
                     !AudioResamplerIA::sampleRateSupported(sampleRate, devSampleRate)) {
                     delete resampler;
-                    resampler = AudioResampler::create(format, channelCount, devSampleRate);
+                    resampler = AudioResampler::create(format, channels, devSampleRate);
+                    resampler->setLocalTimeFreq(sLocalTimeFreq);
                 }
             }
 #endif
-- 
2.7.4

